### PR TITLE
Issue16; Finalizing Lmod .specs for production usage

### PR DESCRIPTION
**Pull request by johnnydevaprasad**

### CODE REVIEW

File: Lmod/branches/7.0/z01-default_modules.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+if [ -z "$__Init_Default_Modules" -o -z "$LD_LIBRARY_PATH" ]; then
+   export __Init_Default_Modules=1;
+   if [ -f "/usr/share/modulefiles/DefaultModules.lua" ];then
+     if [ -z "$LMOD_SYSTEM_DEFAULT_MODULES" ];then
+       export LMOD_SYSTEM_DEFAULT_MODULES="DefaultModules"
```
Comment (fgeorgatos) [inline Lmod/branches/7.0/z01-default_modules.sh]: Hi, 1) `/usr/share/modulefiles/DefaultModules.lua` does not necessarily exist across all systems,
`LMOD_SYSTEM_DEFAULT_MODULES` should be tunable regardless of that.
2) there is missing configuration for .csh users; we don't have many of them but we need not have them complain either

Comment (johnnydevaprasad) [inline Lmod/branches/7.0/z01-default_modules.sh]: Hi Fotis, can you take a look at the next 3 commits.

Comment (fgeorgatos) [inline Lmod/branches/7.0/z01-default_modules.sh]: OK, point (2) is addressed I guess now

still need to understand how we overcome this:

```
[fgeorgatos@node001 ~]$ ll /usr/share/modulefiles/DefaultModules.lua
ls: cannot access /usr/share/modulefiles/DefaultModules.lua: No such file or directory
```

**Comment by fgeorgatos** [comment]
3 APs for today:

1) let's reverse conditionals on L3/L4 (lmod initializations)
2) get closer in implementation to: https://bodhi.fedoraproject.org/updates/FEDORA-2016-8ce6149e81
or, even better 6.3.1-3 specfile for f23 Lmod implementation
3) provide a (commented) hint for `export LMOD_PACKAGE_PATH=/etc/site/lmod`

File: Lmod/branches/7.0/z01-default_modules.sh (hunk 1 above)
Comment (fgeorgatos) [inline Lmod/branches/7.0/z01-default_modules.sh]: fyi. discussed: reverse L3/L4 conditionals, and add an `else` clause specifying `StdEnv` if all other conditions fail!

**Comment by fgeorgatos** [comment]
so, not only Fedora, also centos7-epel-testing carries now the bugfix against overwriting `$MODULEPATH`:
https://pkgs.org/centos-7/epel-testing-x86_64/Lmod-6.3.1-1.el7.x86_64.rpm.html

File: Lmod/branches/7.0/z01-default_modules.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+if [ -z "$__Init_Default_Modules" -o -z "$LD_LIBRARY_PATH" ]; then
+   export __Init_Default_Modules=1;
+   if [ -z "$LMOD_SYSTEM_DEFAULT_MODULES" ];then
+     if [ -f "/usr/share/modulefiles/DefaultModules.lua" ];then
+       export LMOD_SYSTEM_DEFAULT_MODULES="DefaultModules"
+     else
+       export LMOD_SYSTEM_DEFAULT_MODULES="StdEnv"
+     fi
+   fi
```
Comment (fgeorgatos) [inline Lmod/branches/7.0/z01-default_modules.sh]: OK, this looks really good implementation allowing all different sites to implement their own customizations in any of 3 different possible ways... very well done!

Do we wish to merge this PR and address the other 2 issues in another branch? or, wait for this one?

**Comment by fgeorgatos** [comment]
commit 8d17d33 reviewed, all good!

**Comment by fgeorgatos** [comment]
commit b4d21e5 reviewed, as discussed, great!

**Comment by fgeorgatos** [comment]
edd9d19 & f4d2f1d reviewed and appear OK... going to merge

File: Lmod/branches/7.0/000-user_is_root.csh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/csh
+
+set _id_=`id -u`
+
+if ( "$_id_" == 0 ) then
+  if ( ! $?USER_IS_R00T ) then
```
Comment (fgeorgatos) [inline Lmod/branches/7.0/000-user_is_root.csh:6]: as discussed, there is a fancy bug above, bugfix PR upcoming...

**Comment by fgeorgatos** [comment]
this was meant to fix #16, yet it requires also #19 for things to be correct/proper

**Pull request merged**